### ISSUE TITLE
To learn how to use Screenplay with SpecFlow, devs must be able to read a tutorial

### ISSUE THREAD

**Issue by craigfowler** [csf-dev/CSF.Screenplay#155]
[The tutorial page for SpecFlow](https://github.com/csf-dev/CSF.Screenplay/wiki/SpecFlowTutorial) is yet to be written.

**Comment by john-at-digirati** [comment]
Hi @craigfowler ,
I was hoping to use your CSF.Screenplay code with Specflow for regression testing but without any documentation its quite difficult, especially since I've had a myriad of issues trying to get it running e.g. not recognising tests (even when following the NUnit tutorial). Any chance of documentation or a tutorial that demonstrates how to use the Screenplay pattern with your library?
Thanks.

**Comment by craigfowler** [comment]
I've been really bad with this library and seriously neglected the documentation.  I also haven't done much dev on my open source projects recently (other life priorities over the past few months).  But - I haven't forgotten this and I will come back to it someday.

In the mean-time, you might be able to [harvest some information from the tests](https://github.com/csf-dev/agiil/tree/v0.9.0/Tests/Agiil.Tests.BDD) in another project of mine.  That's pretty much why I developed Screenplay, so that I could use it for regression testing, and over there, I do use SpecFlow.

That linked area has my `.feature` files, and the SpecFlow bindings, as well as Screenplay actions etc are found in the [Agiil.BDD](https://github.com/csf-dev/agiil/tree/v0.9.0/Tests/Agiil.BDD) project.

There's also [a screenplay integration config](https://github.com/csf-dev/agiil/blob/v0.9.0/Tests/Agiil.Tests.BDD/ScreenplayIntegrationConfig.cs) which will have some relevant info.

In the absence of real docs, you might be able to use these examples to get going?

**Comment by john-at-digirati** [comment]
Thanks Craig,
I'll use the links you provided.

**Comment by craigfowler** [comment]
One thing I can offer, as the benefit of hindsight, is that I think I went overboard on the acceptance tests. I left my testing pyramid looking a bit like a testing hourglass.

I'm planning on using *SpecFlow tags* to mark up my features, so that I can convert the majority of test cases into simpler integration tests, and not full end-to-end browser tests.

I may or may not end up using screenplay for the converted ones.

**Comment by craigfowler** [comment]
*I have just noticed/realised that I had the wiki permissions set incorrectly.*  Nobody would have been able to contribute to it except myself.

I have just relaxed them, so that others may contribute if they wish.  In other news I am just starting to get geared up again for spare-time projects.  In reality it may still be quite some time before I pick them up again in earnest, but I can at least see my return on the horizon now.
